akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -922,6 +934,53 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       }
     }
 
+    "enforce settings" should {
+
+      "reject new substreams when exceeding SETTINGS_MAX_CONCURRENT_STREAMS" in new TestSetup with RequestResponseProbes {
```

Comment:
We can remove the TODO test `received SETTINGS_MAX_CONCURRENT_STREAMS should limit the number of streams` on line 915 now ;)

Reply:
Ah! No, we can't remove that test! ;-)
That test is about respecting remote settings from _this_ peer.

Note this PR is only about sending the setting and enforcing it on incoming streams. We should use `enforce` and `respect` carefully.

Bonus: the test in line 915 (respect the client-set `SETTINGS_MAX_CONCURRENT_STREAMS` from the server) only makes sense once we add support for PUSH_PROMISE.

Reply:
Ah you're right of course :+1: